### PR TITLE
controller(ticdc): add query interface to controller

### LINKED ISSUE #9395: controller(ticdc): add query interface to controller
add query interface to controller

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #9395

### What is changed and how it works?
add query interface for the controller

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/api/middleware/middleware.go [hunk 1]
```diff
@@ -83,10 +83,10 @@ func ErrorHandleMiddleware() gin.HandlerFunc {
 // is not owner, or handle it locally.
 func ForwardToOwnerMiddleware(p capture.Capture) gin.HandlerFunc {
```
Comment (3AceShowHand) [inline cdc/api/middleware/middleware.go:84]: this method can be renamed

File: cdc/api/status/status.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ func (h *statusAPI) handleStatus(w http.ResponseWriter, req *http.Request) {
 		}
 
 		st.ID = info.ID
-		st.IsOwner = h.capture.IsOwner()
+		st.IsOwner = h.capture.IsController()
```
Comment (3AceShowHand) [inline cdc/api/status/status.go:84]: `IsOwner` can be renamed to `IsController`

File: cdc/api/util.go [hunk 3]
```diff
@@ -150,8 +150,8 @@ func HandleOwnerScheduleTable(
 	}
 }
 
-// ForwardToOwner forwards an request to the owner
-func ForwardToOwner(c *gin.Context, p capture.Capture) {
+// ForwardToController forwards a request to the owner
```
Comment (3AceShowHand) [inline cdc/api/util.go:153]: ditto

File: cdc/api/v1/api.go [hunk 4]
```diff
@@ -852,7 +852,7 @@ func (h *OpenAPI) ServerStatus(c *gin.Context) {
 		Pid:       os.Getpid(),
 		ID:        info.ID,
 		ClusterID: h.capture.GetEtcdClient().GetClusterID(),
-		IsOwner:   h.capture.IsOwner(),
+		IsOwner:   h.capture.IsController(),
```
Comment (3AceShowHand) [inline cdc/api/v1/api.go:855]: ditto

File: cdc/controller/controller.go [hunk 5]
```diff
@@ -30,6 +30,13 @@ import (
 	"golang.org/x/time/rate"
 )
 
+type controllerJobType int
+
+// All OwnerJob types
```
Comment (3AceShowHand) [inline cdc/controller/controller.go:35]: `ControllerJob` ?

File: cdc/api/util.go [hunk 6]
```diff
@@ -150,8 +150,8 @@ func HandleOwnerScheduleTable(
...
+func ForwardToController(c *gin.Context, p capture.Capture) {
 	ctx := c.Request.Context()
 	// every request can only forward to owner one time
```
Comment (CharlesCheung96) [inline cdc/api/util.go:156]:
```suggestion
	// every request can only forward to controller one time
```

File: cdc/api/v2/controller.go [hunk 7]
```diff
@@ -19,16 +19,16 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
-// ResignOwner makes the current owner resign
-// @Summary Notify the owner to resign
+// ResignController makes the current controller resign
+// @Summary Notify the controller to resign
 // @Description Notify the current owner to resign
```
Comment (CharlesCheung96) [inline cdc/api/v2/controller.go:24]:
```suggestion
// @Description Notify the current controller to resign
```

File: cdc/api/v2/health.go [hunk 8]
```diff
@@ -29,7 +29,7 @@ import (
 // @Failure 500,400 {object} model.HTTPError
 // @Router	/api/v2/health [get]
 func (h *OpenAPIV2) health(c *gin.Context) {
-	if !h.capture.IsOwner() {
+	if !h.capture.IsController() {
 		middleware.ForwardToOwnerMiddleware(h.capture)(c)
```
Comment (CharlesCheung96) [inline cdc/api/v2/health.go:33]:
```suggestion
		middleware.ForwardToControllerMiddleware(h.capture)(c)
```

File: cdc/controller/query.go [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+// Query queries owner internal information.
+func (o *controllerImpl) Query(query *Query, done chan<- error) {
+	o.pushOwnerJob(&controllerJob{
+		Tp:    controllerJobTypeQuery,
+		query: query,
+		done:  done,
+	})
+}
+
+func (o *controllerImpl) pushOwnerJob(job *controllerJob) {
```
Comment (CharlesCheung96) [inline cdc/controller/query.go:79]:
```suggestion
func (o *controllerImpl) pushControllerJob(job *controllerJob) {
```

File: cdc/api/middleware/middleware.go (hunk 1 above)
Comment (sdojjy) [reply cdc/api/middleware/middleware.go]: done

File: cdc/controller/controller.go (hunk 5 above)
Comment (sdojjy) [reply cdc/controller/controller.go]: done

File: cdc/api/util.go (hunk 6 above)
Comment (sdojjy) [reply cdc/api/util.go]: done

File: cdc/api/v2/controller.go (hunk 7 above)
Comment (sdojjy) [reply cdc/api/v2/controller.go:24]: done

File: cdc/api/v2/health.go (hunk 8 above)
Comment (sdojjy) [reply cdc/api/v2/health.go]: done

File: cdc/controller/query.go (hunk 9 above)
Comment (sdojjy) [reply cdc/controller/query.go]: done

**Review by 3AceShowHand — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**